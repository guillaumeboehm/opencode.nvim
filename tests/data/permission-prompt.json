[
  {
    "type": "server.connected",
    "properties": {}
  },
  {
    "type": "message.updated",
    "properties": {
      "info": {
        "role": "assistant",
        "time": {
          "created": 1760588856294
        },
        "mode": "plan",
        "path": {
          "cwd": "/Users/cam/Dev/neovim-dev/opencode.nvim",
          "root": "/Users/cam/Dev/neovim-dev/opencode.nvim"
        },
        "cost": 0,
        "modelID": "claude-sonnet-4.5",
        "providerID": "github-copilot",
        "tokens": {
          "cache": {
            "read": 0,
            "write": 0
          },
          "reasoning": 0,
          "input": 0,
          "output": 0
        },
        "id": "msg_9eb45fbe60020xE560OGH3Vdoo",
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry"
      },
      "parts": [
        {
          "id": "prt_9eb46047e001a0A4qdfT2M7iK1",
          "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo",
          "type": "step-start",
          "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry"
        },
        {
          "type": "text",
          "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
          "time": {
            "start": 1760588862279,
            "end": 1760588862279
          },
          "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
          "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the extmark namespace:",
          "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
        },
        {
          "state": {
            "input": {
              "description": "Find extmark namespace usage",
              "command": "rg \"nvim_buf_get_extmarks|ns_id\" /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua -B 2 -A 2"
            },
            "status": "running",
            "time": {
              "start": 1760588861851
            }
          },
          "type": "tool",
          "id": "prt_9eb460c96001AONHjoXQlZBX1c",
          "tool": "bash",
          "callID": "toolu_vrtx_014myMV25GuNHmBhzL8czw1z",
          "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
          "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
        }
      ]
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "id": "prt_9eb46047e001a0A4qdfT2M7iK1",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo",
        "type": "step-start",
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understan",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have ext",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `v",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text`",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`).",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understan",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the extmark namespace",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588858562
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the extmark namespace:",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "state": {
          "status": "pending"
        },
        "type": "tool",
        "id": "prt_9eb460c96001AONHjoXQlZBX1c",
        "tool": "bash",
        "callID": "toolu_vrtx_014myMV25GuNHmBhzL8czw1z",
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "permission.updated",
    "properties": {
      "time": {
        "created": 1760588861851
      },
      "title": "rg \"nvim_buf_get_extmarks|ns_id\" /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua -B 2 -A 2",
      "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo",
      "type": "bash",
      "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
      "metadata": {
        "patterns": [
          "rg /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua *"
        ],
        "command": "rg \"nvim_buf_get_extmarks|ns_id\" /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua -B 2 -A 2"
      },
      "id": "per_9eb46119b001UzsqMdsOXfX0Ij",
      "callID": "toolu_vrtx_014myMV25GuNHmBhzL8czw1z",
      "pattern": [
        "rg /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua *"
      ]
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "state": {
          "input": {
            "description": "Find extmark namespace usage",
            "command": "rg \"nvim_buf_get_extmarks|ns_id\" /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua -B 2 -A 2"
          },
          "status": "running",
          "time": {
            "start": 1760588861851
          }
        },
        "type": "tool",
        "id": "prt_9eb460c96001AONHjoXQlZBX1c",
        "tool": "bash",
        "callID": "toolu_vrtx_014myMV25GuNHmBhzL8czw1z",
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  },
  {
    "type": "message.part.updated",
    "properties": {
      "part": {
        "type": "text",
        "id": "prt_9eb4604c2001bqoqmvAsSXl2Bs",
        "time": {
          "start": 1760588862279,
          "end": 1760588862279
        },
        "sessionID": "ses_6157e31a8ffeqjFIL7oor5c3Ry",
        "text": "Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the extmark namespace:",
        "messageID": "msg_9eb45fbe60020xE560OGH3Vdoo"
      }
    }
  }
]
