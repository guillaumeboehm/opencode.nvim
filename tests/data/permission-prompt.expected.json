{"timestamp":1761241892,"lines":["","----","","","Perfect! Now I understand how it works. The message headers have extmarks with `virt_text` where the first element contains the icon (either `header_user` or `header_assistant`). Let me check the output_window module to understand the extmark namespace:","","**  run** `Find extmark namespace usage`","","```bash","> rg \"nvim_buf_get_extmarks|ns_id\" /Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/output_window.lua -B 2 -A 2","","```","","> [!WARNING] Permission required to run this tool.",">","> Accept `a`    Always `A`    Deny `d`",""],"extmarks":[[1,2,0,{"right_gravity":true,"priority":10,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":false,"virt_text_win_col":-3,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["PLAN","OpencodeMessageRoleAssistant"],[" claude-sonnet-4.5","OpencodeHint"],[" (2025-10-16 04:27:36)","OpencodeHint"],[" [msg_9eb45fbe60020xE560OGH3Vdoo]","OpencodeHint"]],"virt_text_pos":"win_col"}],[2,6,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[3,7,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[4,8,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[5,9,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[6,10,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[7,11,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[8,12,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[9,13,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[10,14,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[11,15,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}],[12,16,0,{"right_gravity":true,"priority":4096,"virt_text_hide":false,"ns_id":3,"virt_text_repeat_linebreak":true,"virt_text_win_col":-1,"virt_text":[["▌","OpencodeToolBorder"]],"virt_text_pos":"win_col"}]],"actions":[]}